### ISSUE TITLE
Question about the Extraction of SEG Token

### ISSUE THREAD

**Issue by c1ircle** [QuentinFitteRey/VLMSAM#9]
Thank you for your excellent work!

I have a question regarding the extraction of the SEG token, as seen here: https://github.com/QuentinFitteRey/VLMSAM/blob/760586da88b9eac7a5b86234d1e64b512d4b66e7/model/vlmsam_generate.py#L144. When training with the forward function of vlm model, it seems that the SEG token does not require shifting by one position.

However, in the implementation of forward of Qwen2_5_VLForConditionalGeneration, there is a shift operation used when calculating the logits and labels:
shift_logits = logits[..., :-1, :].contiguous()
shift_labels = labels[..., 1:].contiguous()
This typically shifts tokens by one position when computing logits and labels, but I am unclear why this shift does not apply to the SEG token extraction process.

Could you clarify why the SEG token extraction does not involve the same shift operation? Is there a particular reason for this difference?

Thank you in advance for your help!
